app/src/main/java/com/eveningoutpost/dexdrip/insulin/LinearTrapezoidInsulin.java

```diff
@@ -37,18 +37,18 @@ public double calculateIOB(double t) {
         else if ((t1 <= time) && (time < t2))
             return 1.0 + 0.5 * max * t1 - max * time;
         else if ((t2 <= time) && (time < t3))
-            return 0.5 * (time - t3) * (time - t3) * max / (t3 - t2);
+            return 0.5 * (t3 - time) * (t3 - time) * max / (t3 - t2);
         else return 0;
     }
 
     public double calculateActivity(double t) {
         double time = t - onset;
         if ((0 <= time) && (time < t1))
-            return concentration * time * max / t2;
+            return concentration * time * max / t1;
         else if ((t1 <= time) && (time < t2))
             return concentration * max;
         else if ((t2 <= time) && (time < t3))
-            return concentration * (time - t3) * max / (t3 - t2);
+            return concentration * (t3 - time) * max / (t3 - t2);
```

Comment:
as we take activity in abs() this change seems useless

Reply:
Two points
1. Why is that `abs()` there anyway?
2. What this function returns in this case is **wrong**. The fact that later in the code, a wrong value is fixed does not make it any less wrong. This is medical software that people trust their health to, we should have a much much higher standard than "it's ok, the mistakes cancel each other out".